commands/errors.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) OpenFaaS Author(s) 2019. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"strings"
+)
+
+const (
+	// NoTLSWarn Warning thrown when no SSL/TLS is used
+	NoTLSWarn = "WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates."
+)
+
+// checkTLSInsecure checks if the given gateway has https. Use tsInsecure to skip validations
+// Note: This function does not stop the flow only prints a warning
+func checkTLSInsecure(gateway string, tlsInsecure bool) string {
```

Comment:
You could add the print statement here and make it testable by using the `io.Writer` interface.  It would look like this

```go
func warnTLSInsecure(out io.Writer, gateway string, tlsInsecure bool) {
	if !tlsInsecure {
		if !strings.HasPrefix(gateway, "https") {
			fmt.Fprintln(out, NoTLSWarn)
		}
	}
}
```

All of the usages would then look like 

```go
warnTLSInsecure(cmd.OutOrStdout(), gatewayAddress, tlsInsecure)
```

No need for print statements.

This is still testable because the `bytes.Buffer` implements `io.Writer`, so the test look like

```go
buf := &bytes.Buffer{}

warnTLSInsecure(buf, tt.args.gateway, tt.args.tlsInsecure)
got := buf.String()

if got != tt.want {
    t.Errorf("checkTLSInsecure() = %v, want %v", got, tt.want)
}
```

Also, important, Cobra commands also have `cmd.OutOrStderr()` which behaves the same way but the output goes to stderr, as implied.  These helpers are useful because it ensures that you get the right output buffers, even if it is being configured to go somewhere else by the user in someway.

Reply:
I agree with this, but would see it as part of a follow-up change. The scope of this PR is only to move the location of the warnings. @mario-mazo is also a new contributor to the project so if we work together more incrementally I think it would make the experience smoother?

Reply:
i agree, we can definitely take this as next steps.  The other three comments are smaller and would be good changes though, I think.

Reply:
I was not aware of `cmd.OutOrStdout()` I will implement this

Reply:
Please don't.

This PR is only to move the message, not to refactor the code. Do that in a follow-up PR.